source/fundamentals/gridfs.txt

```diff
@@ -1,86 +1,84 @@
-.. uses gridfs.rst
+.. _pymongo-gridfs:
 
-GridFS Example
-==============
+GridFS
+======
 
-.. code-block:: python
-
-  from pymongo import MongoClient
-
-  client = MongoClient()
-  client.drop_database("gridfs_example")
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
 
-This example shows how to use ``gridfs`` to store large binary
-objects (e.g. files) in MongoDB.
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: binary large object, blob, file, storage
 
-.. seealso:: The API docs for ``gridfs``.
-
-.. seealso:: `This blog post
-   <http://dirolf.com/2010/03/29/new-gridfs-implementation-for-pymongo.html>`_
-   for some motivation behind this API.
+This guide shows how to use GridFS to store and retrieve large
+in MongoDB.
 
 Setup
 -----
 
-We start by creating a ``~gridfs.GridFS`` instance to use:
+Each GridFS instance operates on a specific MongoDB database.
+To create a ``GridFS`` instance, call the ``GridFS()``
+constructor and pass in a MongoDB database:
 
 .. code-block:: python
 
-  >>> from pymongo import MongoClient
-  >>> import gridfs
-  >>>
-  >>> db = MongoClient().gridfs_example
-  >>> fs = gridfs.GridFS(db)
-
-Every ``~gridfs.GridFS`` instance is created with and will
-operate on a specific ``~pymongo.database.Database`` instance.
+   >>> from pymongo import MongoClient
+   >>> import gridfs
+   >>>
+   >>> db = MongoClient().gridfs_example
+   >>> fs = gridfs.GridFS(db)
 
 Saving and Retrieving Data
 --------------------------
 
-The simplest way to work with ``gridfs`` is to use its key/value
-interface (the the ``~gridfs.GridFS.put`` method and
-the ``~gridfs.GridFS.get`` method methods). To write data to GridFS, use
-the ``~gridfs.GridFS.put`` method:
+GridFS offers a key-value-pair interface through the ``~gridfs.GridFS.put()`` and
+``~gridfs.GridFS.get()`` methods.
+
+To write data to GridFS, use the ``~gridfs.GridFS.put()`` method:
 
 .. code-block:: python
 
-  >>> a = fs.put(b"hello world")
+   >>> a = fs.put("hello world")
 
-the ``~gridfs.GridFS.put`` method creates a new file in GridFS, and returns
-the value of the file document's ``"_id"`` key. Given that ``"_id"``
-we can use the ``~gridfs.GridFS.get`` method to get back the contents of the
-file:
+The ``~gridfs.GridFS.put()`` method creates a new file in GridFS, and returns
+the value of the file document's ``_id`` key. You can use the file's returned ``_id``
```

Comment:
Q: Is "file document" a GridFS term? Or should it be either "file" or "document" (referring to the "value of the file document's ... " part here)?

Reply:
Good question. This is the language of the original page, but our other docs just say "file", so that's what I'll use here.